puppet/modules/prometheus/manifests/init.pp

```diff
@@ -6,15 +6,19 @@
   Integer[1025,65535] $etcd_k8s_main_port = $::prometheus::params::etcd_k8s_main_port,
   Integer[1025,65535] $etcd_k8s_events_port = $::prometheus::params::etcd_k8s_events_port,
   Integer[1024,65535] $etcd_overlay_port = $::prometheus::params::etcd_overlay_port,
+  Boolean $external_scrape_targets_only = false,
 ) inherits ::prometheus::params
 {
 
   if $role == 'master' {
     include ::prometheus::server
-    include ::prometheus::kube_state_metrics
-    include ::prometheus::node_exporter
-    include ::prometheus::blackbox_exporter
     include ::prometheus::blackbox_exporter_etcd
+    include ::prometheus::node_exporter
+    if ! $external_scrape_targets_only {
+      include ::prometheus::kube_state_metrics
+      include ::prometheus::node_exporter
+      include ::prometheus::blackbox_exporter
+    }
```

Comment:
node_exporter is included in both cases, is this intended?

Reply:
Fixed in latest push